### PR TITLE
Update ory charts

### PR DESCRIPTION
**Pull request by Demonsthere**
## Proposed changes

- Change master to oathkeeper-master subchart
- Fix some small errors in all charts
- Unify common values across all charts
- Change hydra secret behaviour:
  - If no secret is given, generate a random one
  - Move secret creation to a pre-install hook, so it won't be touched during upgrades
- Update docs for all charts

## Checklist

- [x] I have read the [contributing guidelines](CONTRIBUTING.md)
- [x] I have read the [security policy](SECURITY.md)
- [x] I confirm that this pull request does not address a security vulnerability. If this pull request addresses a security
vulnerability, I confirm that I got green light (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation within the code base (if appropriate)

### CODE REVIEW

File: helm/charts/oathkeeper/charts/oathkeeper-master/templates/deployment.yaml [hunk 1]
```diff
@@ -41,7 +39,7 @@ spec:
             - /manager
           args:
             - --metrics-addr=127.0.0.1:8080
-            - --rulesConfigmapName={{ .Values.configMapName }}
+            - --rulesConfigmapName={{ .Release.Name }}-rules
```
Comment (aeneasr) [inline helm/charts/oathkeeper/charts/oathkeeper-master/templates/deployment.yaml]: I think this needs to be `oathkeeper.fullname` right?

Comment (Demonsthere) [reply helm/charts/oathkeeper/charts/oathkeeper-master/templates/deployment.yaml]: no, as this is a subchart, it inherits the name of its parent, so `fullname` produces `oathkeeper-oathkeeper-maester`

Comment (Demonsthere) [reply helm/charts/oathkeeper/charts/oathkeeper-master/templates/deployment.yaml]: After some thought I changes this to a smarter function

File: helm/charts/hydra/templates/secrets.yaml [hunk 2]
```diff
@@ -5,8 +5,13 @@ metadata:
...
+  annotations:
+    # Create the secret before installation, and only then. This saves the secret from regenerating during an upgrade
+    "helm.sh/hook": "pre-install"
+    "helm.sh/hook-delete-policy": "before-hook-creation"
 type: Opaque
 data:
-  secretsSystem: {{ include "hydra.secrets.system" . | required "Value secrets.system can not be empty!" | b64enc | quote }}
-  secretsCookie: {{ include "hydra.secrets.cookie" . | required "Value secrets.cookie can not be empty!" | b64enc | quote }}
+  # Generate a random secret if the user doesn't give one. User given password has priority
+  secretsSystem: {{ ( include "hydra.secrets.system" . | default ( randAlphaNum 32 )) | required "Value secrets.system can not be empty!" | b64enc | quote }}
```
Comment (aeneasr) [inline helm/charts/hydra/templates/secrets.yaml]: Is this cryptographically secure?

File: helm/charts/oathkeeper/values.yaml [hunk 3]
```diff
@@ -114,3 +114,7 @@ deployment:
   annotations: {}
 # Configure node affinity
 affinity: {}
+
+# Configures controller setup
+oathkeeper-maester:
```
Comment (aeneasr) [inline helm/charts/oathkeeper/values.yaml]: I think helm convention is camelCase

Comment (Demonsthere) [reply helm/charts/oathkeeper/values.yaml]: [Naming convention](https://helm.sh/docs/chart_best_practices/#chart-names) names with `-` are fine too, as long as the chart resided in a dir with the same name :)

File: helm/charts/hydra/templates/secrets.yaml (hunk 2 above)
Comment (Demonsthere) [reply helm/charts/hydra/templates/secrets.yaml]: what do you mean by that?

Comment (aeneasr) [reply helm/charts/hydra/templates/secrets.yaml]: Are values coming from `randAlphaNum` cryptographically secure? Meaning that they come from e.g. `/dev/urandom` as opposed to a PRNG

Comment (Demonsthere) [reply helm/charts/hydra/templates/secrets.yaml]: Took me a while but found the [docs](https://github.com/Masterminds/sprig/blob/40a817194fa6bb2cd370b302c30b27d3706f122d/docs/strings.md#randalphanum-randalpha-randnumeric-and-randascii) for this function, seems we are safe :)

Comment (aeneasr) [reply helm/charts/hydra/templates/secrets.yaml]: Awesome!

**Comment by aeneasr** [comment]
I think this looks pretty solid! Maybe add some docs to oathkeeper and hydra to show the new changes?

**Comment by Demonsthere** [comment]
> This looks really solid!! My last thought is if we should rename the config setting `oathkeeper-maester.enabled` to `oathkeeper.maester.enabled` (so make it a subkey of oathkeeper instead of a top level thing). It would fit in more nicely with the existing config API. Right now, the hyphen stands very lonely as we're not using hyphens anywhere else in the helm charts.

As much as I agree that it stand out like a sore thumb, the key has to reflect the name of the subchart (which is `oathkeeper-maester`). We would need to use some aliases in order to change that

**Comment by aeneasr** [comment]
Wouldn't it be enough to change https://github.com/ory/k8s/pull/37/files#diff-6784805510d68d1677ccaf9f3b5e05c9R4 to:

```
     condition: oathkeeper.maester.enabled
```

?

**Pull request merged**